src/features/mail/components/Inbox.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useContext } from "react";
+import { Accordion } from "react-bootstrap";
+
+import { InnerPanel, OuterPanel } from "components/ui/Panel";
+import { Context } from "../MailProvider";
+
+interface Props {
+}
+
+/**
+ * TODO
+ * - markdown message body?
+ */
+export const Inbox: React.FC<Props> = () => {
+  const { inbox, setRead } = useContext(Context);
```

Comment:
Do we need a provider to maintain this state? The Provider pattern is useful when you have multiple children sharing the same state. In this case here we only have 1 component handling this state.

I would suggest moving the state from the `MailProvider` into this component so it is more encapsulated.

Reply:
Do you mean completely removing `MailProvidertsx` and put all its contents in `Inbox.tsx` ? I used Provider since `Mail.tsx` uses `hasUnread` prop which has dependency from `inbox` to show/hide alerted expression. Going with that, I exposed `setRead` to edit the unread state from inbox array locally. 

Or would it be better to put `MailProvider` contents in `Mail` then pass props to `Inbox`?

Reply:
@j923sneaks Yes, I think moving the contents of MailProvider into Mail and passing via props is ideal. 

That will keep the `Inbox.tsx` a controlled component (a dumb UI component without much business logic)

The following is an example of how these props could look:

```
interface Props {
  inbox: Message[]
  onRead: (index: number) => void
}
```

These are just a suggestion to help make the components a bit more focussed (Not necessary for approval) :)